api/client/client_test.go

```diff
@@ -104,7 +104,7 @@ func startMockServerWithListener(t *testing.T, l net.Listener) *mockServer {
 	t.Cleanup(srv.grpc.Stop)
 
 	go func() {
-		require.NoError(t, srv.grpc.Serve(l))
+		srv.grpc.Serve(l)
```

Comment:
FYI, you can use assert.NoError - the problem is calling `t.Fatal()` (directly or indirectly) in goroutines, calling `t.Error` (and variants) is alright.

In this case I think keeping the assert is good, so we can narrow down mysterious failures to gRPC not starting.

```suggestion
		assert.NoError(t, srv.grpc.Serve(l))
```

Reply:
Yeah, though it's still racy, if the main test goroutine finishes before this one, the assert will be missed and an error from `Serve` won't fail the test. The proper way to catch this if you really care about `Serve` not returning an error is to dump the result to a channel and check it when the test finishes, but that's hard to do with an asynchronous helper like this.

In this case, I figured if the server fails to start, the test will fail on a future assertion.

Reply:
> In this case, I figured if the server fails to start, the test will fail on a future assertion

I think this is exactly when it makes a difference, especially when one goes about debugging a test they haven't seen before. Up to you.